CrossrefReferenceLinkingPlugin.php

```diff
@@ -0,0 +1,474 @@
+	{
+		// If crossref plugin is set i.e. the crossref credentials exist we can assume that DOI plugin is set correctly
+		$crossrefPlugin = PluginRegistry::getPlugin('generic', 'crossrefplugin');
+		return $crossrefPlugin && $crossrefPlugin->getSetting($contextId, 'username') && $crossrefPlugin->getSetting($contextId, 'password');
+	}
+
+	/**
+	 * Are citations submission metadata enabled in this journal
+	 */
+	public function citationsEnabled(int $contextId): bool
+	{
+		$contextDao = Application::getContextDAO();
+		$context = $contextDao->getById($contextId);
+		return !empty($context->getSetting('citations'));
+	}
+
+	/**
+	 * @copydoc Plugin::getDisplayName()
+	 */
+	public function getDisplayName()
+	{
+		return __('plugins.generic.crossrefReferenceLinking.displayName');
+	}
+
+	/**
+	 * @copydoc Plugin::getDescription()
+	 */
+	public function getDescription()
+	{
+		return __('plugins.generic.crossrefReferenceLinking.description');
+	}
+
+	/**
+	 * @see Plugin::getActions()
+	 */
+	public function getActions($request, $actionArgs)
+	{
+		$actions = parent::getActions($request, $actionArgs);
+		if (!$this->getEnabled()) {
+			return $actions;
+		}
+		$router = $request->getRouter();
+		$linkAction = new LinkAction(
+			'settings',
+			new AjaxModal(
+				$router->url(
+					$request,
+					null,
+					null,
+					'manage',
+					null,
+					array(
+						'verb' => 'settings',
+						'plugin' => $this->getName(),
+						'category' => 'generic'
+					)
+				),
+				$this->getDisplayName()
+			),
+			__('manager.plugins.settings'),
+			null
+		);
+		array_unshift($actions, $linkAction);
+		return $actions;
+	}
+
+	/**
+	 * @copydoc Plugin::manage()
+	 */
+	public function manage($args, $request)
+	{
+		$context = $request->getContext();
+		switch ($request->getUserVar('verb')) {
+			case 'settings':
+				$form = new CrossrefReferenceLinkingSettingsForm($this, $context->getId());
+				$form->initData();
+				return new JSONMessage(true, $form->fetch($request));
+			case 'save':
+				$form = new CrossrefReferenceLinkingSettingsForm($this, $context->getId());
+				$form->readInputData();
+				if ($form->validate()) {
+					$form->execute($request);
+					$notificationManager = new NotificationManager();
+					$notificationManager->createTrivialNotification(
+						$request->getUser()->getId(),
+						Notification::NOTIFICATION_TYPE_SUCCESS,
+						array('contents' => __('plugins.generic.crossrefReferenceLinking.settings.form.saved'))
+					);
+					return new JSONMessage(true);
+				}
+				return new JSONMessage(true, $form->fetch($request));
+		}
+		return parent::manage($args, $request);
+	}
+
+	/**
+	 * @see AcronPlugin::_parseCrontab()
+	 *
+	 * @param $hookName string 'AcronPlugin::parseCronTab'
+	 * @param $args array [
+	 *  @option array Task files paths
+	 * ]
+	 */
+	public function callbackParseCronTab(string $hookName, array $args): bool
+	{
+		if ($this->getEnabled() || !Config::getVar('general', 'installed')) {
```

Comment:
I think you can use the `Application::isUnderMaintenance()` instead of the `Config::getVar()`.